src/Estimators/include/BipedalLocomotion/RobotDynamicsEstimator/SubModel.h

```diff
@@ -0,0 +1,273 @@
+/**
+ * @file SubModel.h
+ * @authors Ines Sorrentino
+ * @copyright 2023 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the BSD-3-Clause license.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_ESTIMATORS_SUBMODEL_H
+#define BIPEDAL_LOCOMOTION_ESTIMATORS_SUBMODEL_H
+
+#include <memory>
+#include <string>
+#include <vector>
+
+// iDynTree
+#include <iDynTree/KinDynComputations.h>
+#include <iDynTree/Model/Model.h>
+#include <iDynTree/ModelIO/ModelLoader.h>
+
+// BLF
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+
+namespace BipedalLocomotion
+{
+namespace Estimators
+{
+namespace RobotDynamicsEstimator
+{
+
+/**
+ * Sensor describes the generic sensors of the submodels.
+ */
+struct Sensor
+{
+    std::string name; /**< Name of the sensor in the URDF */
+    std::string frame; /**< Frame of the sensor in the URDF */
+};
+
+/**
+ * FT describes the force/torque sensors of the submodels.
+ */
+struct FT : Sensor
+{
+    enum class Direction : int
+    {
+        Positive = 1,
```

Comment:
Probably it may be worth to specify if "positive" means the force that the rest of the system applied on the considered submodel, or that the submodel applies on the rest of the system. "Positive" or "Negative" on their own risk of being a bit ambigous.

Reply:
> Probably it may be worth to specify if "positive" means the force that the rest of the system applied on the considered submodel, or that the submodel applies on the rest of the system. "Positive" or "Negative" on their own risk of being a bit ambiguous.

@traversaro  Yes, it's ambiguous. Are `Compressing` and `Stretching` more explicative? Of course, I'm going to add also the documentation as you suggested.

Reply:
I don't think `Compressing` or `Stretching` are more explicative, as depending on the sign of the measure you can have a sensor that is marked as "Compressing" that is measuring a force that is "Stretching" and the other way around, but this is just my taste.